Don't allow the player to scroll world map while hero is moving

oleg-derevenetz opened the pull request:
fix #1159

Also fixes an issue when cursor isn't turned into Cursor::WAIT when hotkey is used to start the hero's movement.

oleg-derevenetz commented:
Hi @ihhub I removed that small delay before starting scrolling using mouse pointer, if you think that it's needed for something, I'll revert it back.

ihhub commented:
> 
> 
> Hi @ihhub I removed that small delay before starting scrolling using mouse pointer, if you think that it's needed for something, I'll revert it back.

Hi @oleg-derevenetz , please revert it back. We need it to avoid immediate scrolling when player click near the edge of the screen. Otherwise, it's very difficult to to manage.

Branikolog commented on src/fheroes2/game/game_startgame.cpp:
```diff
@@ -950,64 +944,83 @@ int Interface::Basic::HumanTurn( bool isload )
+
+                display.render();
+
+                continue;
```
In the OG animation proceeds, while scrolling.
Stopping it can be noticeable on low scrolling speeds...

oleg-derevenetz replied:
Hi @ihhub this continue is already there in the current version. I thought it is here for a reason. Should I remove it?

oleg-derevenetz replied:
Hi @ihhub @Branikolog I just checked - it doesn't stop the animation of objects. They have time to animate in the period of those SCROLL_DELAY ms as it happens in the current version of the corresponding code.

oleg-derevenetz commented on src/fheroes2/game/game_delays.cpp:
```diff
@@ -66,36 +66,37 @@ namespace Game
-                          0, // CUSTOM_DELAY
-                          0};
+    TimeDelay delays[] = { 20, // SCROLL_DELAY
+                           500, // SCROLL_START_DELAY
```
Delay of 20ms is barely noticeable. May be something between?

oleg-derevenetz commented:
Hi @ihhub

> Please reduce this value to 20. RIght now the overall delay takes 1 second.

Well, in fact it's between 500ms and 1s :) But I agree that it may be too long. On the other hand, 20ms is barely noticeable - like no delay at all. I changed it to 250, please try it again, it it still too long, I can change it to 100.

> By doing this we're stopping animation of objects on map while scrolling.

Although it's not really the case - it doesn't stop the animation of objects (they have time to animate in the period of those SCROLL_DELAY ms as it happens in the current version of the corresponding code with the same `continue` after `render()`) it really seems useless, so I removed it.

ihhub commented:
Let's ask an expert in the game:
@LeHerosInconnu, could you please tell what's the delay before starting scrolling in the original game?

oleg-derevenetz commented:
Hi @ihhub BTW I never used consoles, how accurate is the mouse emulation on them in relation to this PR? How likely is it to miss the pointer near edges?

ihhub commented:
> Hi @ihhub BTW I never used consoles, how accurate is the mouse emulation on them in relation to this PR? How likely is it to miss the pointer near edges?

I don't have any consoles to test but @Northfear can shed a light on this question.

LeHerosInconnu commented:
Hello @ihhub,

> Let's ask an expert in the game:
> @LeHerosInconnu, could you please tell what's the delay before starting scrolling in the original game?

In the original game, there is no noticeable delay.

In original game:

https://user-images.githubusercontent.com/43583994/115145806-c2735480-a053-11eb-949e-f7f22e01be4c.mp4

oleg-derevenetz commented:
Well, let's make no noticeable delay then :)

ihhub commented:
@oleg-derevenetz , thanks a lot for fixing this bug!
